### ISSUE TITLE
Hi, how can I save a XGBRegressor into a txt file?

### ISSUE THREAD

**Issue by ddghjikle** [dmlc/xgboost#7222]
Hi, I have defined a XGBRegressor as :
import xgboost as xgb
xgb1 = xgb.sklearn.XGBRegressor(
         learning_rate =0.1,
         n_estimators=100,
         max_depth=5,
         min_child_weight=11,
         gamma=0.1,
         subsample=0.8,
         colsample_bytree=0.7,
         objective='reg:squarederror',
         n_jobs=-1,
         scale_pos_weight=1,
         seed=1234)
However, I cannot run the dump_model to save this model as dump or txt files which are human-readable. It also seems that the official document does not mention the dump_model function. Could you please give me some suggestions?

**Comment by trivialfis** [comment]
You can use `get_dump` to obtain a string representation of the model.  But the output cannot be loaded back so it's not exactly "saving the model".  If that's what you want, you can save the string using Python builtin function:

``` python
model = xgb.XGBRegressor()
model.fit(X, y)
text_dump = model.get_booster().get_dump()
with open("model.txt", "w") as fd:
    fd.write(text_dump)
```

If you want a loadable representation, you can use:

``` python
model = xgb.XGBRegressor()
model.fit(X, y)
model.save_model("model.json")
```

**Comment by ddghjikle** [comment]
Hi, thanks very much for your help. Now, i have saved my XGBRegressor into a json file:

{"learner":{"attributes":{"scikit_learn":"{\"n_estimators\": 1, \"objective\": \"reg:squarederror\", \"max_depth\": 5, \"learning_rate\": 0.1, \"verbosity\": null, \"booster\": null, \"tree_method\": null, \"gamma\": 0.1, \"min_child_weight\": 11, \"max_delta_step\": null, \"subsample\": 0.8, \"colsample_bytree\": 0.7, \"colsample_bylevel\": null, \"colsample_bynode\": null, \"reg_alpha\": null, \"reg_lambda\": null, \"scale_pos_weight\": 1, \"base_score\": null, \"missing\": NaN, \"num_parallel_tree\": null, \"kwargs\": {\"seed\": 1234}, \"random_state\": null, \"n_jobs\": -1, \"monotone_constraints\": null, \"interaction_constraints\": null, \"importance_type\": \"gain\", \"gpu_id\": null, \"validate_parameters\": null, \"_estimator_type\": \"regressor\"}"},"feature_names":[],"feature_types":[],"gradient_booster":{"model":{"gbtree_model_param":{"num_trees":"1","size_leaf_vector":"0"},"tree_info":[0],"trees":[{"base_weights":[0E0],"categories":[],"categories_nodes":[],"categories_segments":[],"categories_sizes":[],"default_left":[false],"id":0,"left_children":[-1],"loss_changes":[0E0],"parents":[2147483647],"right_children":[-1],"split_conditions":[0E0],"split_indices":[0],"split_type":[0],"sum_hessian":[1E0],"tree_param":{"num_deleted":"0","num_feature":"150528","num_nodes":"1","size_leaf_vector":"0"}}]},"name":"gbtree"},"learner_model_param":{"base_score":"5E-1","num_class":"0","num_feature":"150528"},"objective":{"name":"reg:squarederror","reg_loss_param":{"scale_pos_weight":"1"}}},"version":[1,4,2]}

However, i am confused by these parameters. Is there an official document can explain these parameters?

**Comment by trivialfis** [comment]
For most of the parameters you can find them here: https://xgboost.readthedocs.io/en/latest/parameter.html

Rest are internal.
